lib/events.js

```diff
@@ -64,11 +68,80 @@ Object.defineProperty(EventEmitter, 'defaultMaxListeners', {
   }
 });
 
+const proxyEventsHandler = {
+  get(emitter, prop) {
+    const events = emitter[kEvents];
+    if (events === undefined)
+      return undefined;
+    return events.get(prop);
+  },
+  set(emitter, prop, value) {
+    let events = emitter[kEvents];
+    if (events === undefined) {
+      events = emitter[kEvents] = new Map();
+      emitter._eventsCount = 0;
+    }
+    events.set(prop, value);
+    return true;
+  },
+  has(emitter, prop) {
+    const events = emitter[kEvents];
+    return events !== undefined && events.has(prop);
+  },
+  getPrototypeOf() {
+    return null;
+  },
+  deleteProperty(emitter, prop) {
+    const events = emitter[kEvents];
+    if (events !== undefined)
+      events.delete(prop);
+    return true;
+  },
+  ownKeys(emitter) {
+    return emitter.eventNames();
+  },
+  defineProperty(emitter, prop, descriptor) {
+    if (!descriptor.writable || !descriptor.configurable)
+      return false;
+    proxyEventsHandler.set(emitter, prop, descriptor.value);
+    return true;
+  },
+  getOwnPropertyDescriptor(emitter, prop) {
+    const value = proxyEventsHandler.get(emitter, prop);
+    if (value === undefined)
+      return undefined;
+    return {
+      value,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    };
+  },
+};
+
+Object.defineProperty(EventEmitter.prototype, '_events', {
+  enumerable: true,
+  get: function() {
+    let proxy = this[kEventsProxy];
+    if (proxy === undefined && this[kEvents] !== undefined)
+      proxy = new Proxy(this, proxyEventsHandler);
+    return proxy;
+  },
+  set: function(value) {
+    if (this[kEvents] === undefined)
+      this.init();
+    if (!value || Array.isArray(value) && value.length === 0) {
+      this[kEvents] = new Map();
+      this._eventsCount = 0;
+    }
```

Comment:
Are we okay with making this a no-op if value is a "real" events object?

Reply:
This looks wrong. Should `value` not be an object with properties? I expect it to look something like:

```js
if (value) {
  this[kEvents] = new Map(Object.entries(value));
  this._eventsCount = this[kEvents].size;
}
```